cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/cube/CubeDeleteQuery.java

```diff
@@ -14,27 +14,37 @@
  * the License.
  */
 
-package co.cask.cdap.metrics.store.cube;
+package co.cask.cdap.api.dataset.lib.cube;
 
+import co.cask.cdap.api.annotation.Beta;
 import com.google.common.base.Joiner;
 import com.google.common.base.Objects;
-import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Maps;
 
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
- * Query that specifies parameters to delete entries from cube.
+ * Defines a query for deleting data in {@link Cube}.
  */
+@Beta
 public class CubeDeleteQuery {
-
   private final long startTs;
   private final long endTs;
   private final int resolution;
   private final String measureName;
   private final Map<String, String> sliceByTagValues;
 
-  public CubeDeleteQuery(long startTs, long endTs, int resolution, String measureName,
+  /**
+   * Creates instance of {@link CubeDeleteQuery} that defines selection of data to delete from {@link Cube}.
+   * @param startTs start time of the data selection, in seconds since epoch
+   * @param endTs end time of the data selection, in seconds since epoch
+   * @param resolution resolution of the aggregations to delete from
+   * @param measureName name of the measure to delete, {@code null} means delete all
+   * @param sliceByTagValues tag name, tag value pairs that define the data selection
+   */
+  public CubeDeleteQuery(long startTs, long endTs, int resolution,
+                         @Nullable String measureName,
```

Comment:
Better to have `@Nullable` argument as the last one. Also, it's handy to have an overloaded constructor to represent delete all so that user doesn't need to pass in `null`.